src/gui/DatabaseOpenWidget.cpp

```diff
@@ -160,15 +188,36 @@ CompositeKey DatabaseOpenWidget::databaseKey()
         }
         masterKey.addKey(key);
         lastKeyFiles[m_filename] = keyFilename;
-    }
-    else {
+    } else {
         lastKeyFiles.remove(m_filename);
     }
 
+    if (m_ui->checkChallengeResponse->isChecked()) {
+        lastChallengeResponse[m_filename] = true;
+    } else {
+        lastChallengeResponse.remove(m_filename);
+    }
+
     if (config()->get("RememberLastKeyFiles").toBool()) {
         config()->set("LastKeyFiles", lastKeyFiles);
     }
 
+#ifdef WITH_XC_YUBIKEY
+    if (config()->get("RememberLastKeyFiles").toBool()) {
+        config()->set("LastChallengeResponse", lastChallengeResponse);
+    }
+
+    if (m_ui->checkChallengeResponse->isChecked()) {
+        int i = m_ui->comboChallengeResponse->currentIndex();
+        i = m_ui->comboChallengeResponse->itemData(i).toInt();
+
+        bool blocking = i & true;
+        int slot      = i >> 1;
+        auto key      = QSharedPointer<YkChallengeResponseKey>(new YkChallengeResponseKey(slot, blocking));
+        masterKey.addChallengeResponseKey(key);
```

Comment:
Since this is in two different classes, it would need to be a public function. It doesn't really belong in either class as a public member (or free function), though.